pkg/sql/tablemetadatacache/util/util.go

```diff
@@ -0,0 +1,16 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package tablemetadatacacheutil
+
+import "context"
+
+// ITableMetadataUpdater is an interface that exposes a RunUpdater
```

Comment:
Why is this defined in the util pkg?

Reply:
This is needed in util because it used in `TestingKnobs` which is also defined in util and would cause a circular dependency otherwise. I tried moving TestingKnobs into tablemetadatacache, but that also introduces a circular dependency of `tablemetadatacache -> sql -> tablemetadatacache` 😞. It'd be nice to not need to depend on the sql package, but it looks like it is needed in order to get `JobExecContext`